airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlCdcIntegrationTest.kt

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql
+
+import io.airbyte.cdk.command.CliRunner
+import io.airbyte.cdk.command.ConfigurationJsonObjectSupplier
+import io.airbyte.cdk.command.ConfiguredCatalogFactory
+import io.airbyte.cdk.command.InputStateFactory
+import io.airbyte.cdk.command.ValidatedJsonUtils
+import io.airbyte.cdk.output.BufferingOutputConsumer
+import io.airbyte.protocol.models.AirbyteStateMessage
+import io.micronaut.context.annotation.Property
+import io.micronaut.context.env.Environment
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import jakarta.inject.Inject
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+@MicronautTest(environments = [Environment.TEST], rebuildContext = true)
+class MysqlCdcIntegrationTest {
+    @Inject
+    lateinit var supplier: ConfigurationJsonObjectSupplier<MysqlSourceConfigurationJsonObject>
+
+    @Inject lateinit var catalogFactory: ConfiguredCatalogFactory
+
+    @Inject lateinit var inputStateFactory: InputStateFactory
+
+    @Test
+    @Property(name = "airbyte.connector.config.json", value = CDC_CONFIG_JSON)
+    @Property(name = "airbyte.connector.catalog.json", value = CDC_CATALOG)
+    fun testReadWithState() {
+        val pojoConfig: MysqlSourceConfigurationJsonObject = supplier.get()
+        val configuredCatalog = catalogFactory.make(CDC_CATALOG)
+        val list: List<io.airbyte.protocol.models.v0.AirbyteStateMessage>? =
+            ValidatedJsonUtils.parseList(
+                    io.airbyte.protocol.models.v0.AirbyteStateMessage::class.java,
+                    AIRBYTE_STATE_MESSAGE_JSON_2
+                )
```

Comment:
You can inject this using `@Property` I think?

Reply:
This test will soon go away - mainly used for the iteration loop. Rodi is working on another integration test here